README.md
[![cran version](http://www.r-pkg.org/badges/version/rcarbon)](https://CRAN.R-project.org/package=rcarbon) 
[![development version](https://img.shields.io/badge/devel%20version-1.5.1-blue.svg)](https://github.com/ahb108/rcarbon)
[![rstudio mirror downloads](http://cranlogs.r-pkg.org/badges/rcarbon?)](https://github.com/metacran/cranlogs.app)
[![rstudio mirror downloads](http://cranlogs.r-pkg.org/badges/grand-total/rcarbon?color=82b4e8)](https://github.com/metacran/cranlogs.app)
[![R-CMD-check](https://github.com/ahb108/rcarbon/workflows/R-CMD-check/badge.svg)](https://github.com/ahb108/rcarbon/actions)
# rcarbon <img src="/logo/rcarbon_logo.png" align="right" />

This package enables the calibration and analysis of radiocarbon dates, often but not exclusively for the purposes of archaeological research. It includes functions not only for basic calibration, uncalibration, and plotting of one or more dates, but also a statistical framework for building demographic and related longitudinal inferences from aggregate radiocarbon date lists, including: Monte-Carlo simulation tests (Timpson et al 2014), random mark permutation tests (Crema et al 2016, Bevan et al 2017), a point-to-point test (Edinborough et al 2017), and spatial permutation tests (Crema, Bevan, and Shennan 2017). For more details see the associated paper (Crema and Bevan 2021).

To install the stable version from CRAN:

```
install.packages('rcarbon')
```

To install the latest development version:

```
# "devtools" package required 
devtools::install_github('ahb108/rcarbon')
```
### Citation
To cite _rcarbon_ in publications use:

Crema, E.R., Bevan, A. 2021 [Inference from Large Sets of Radiocarbon Dates: Software and Methods](https://doi.org/10.1017/RDC.2020.95) Radiocarbon 63(1),23-39  doi:10.1017/RDC.2020.95

### References
Bevan, A. Colledge,S., Fuller, D., Fyfe, R., Shennan, S., Stevens, C. 2017.[Holocene fluctuations in human population demonstrate repeated links to food production and climate](https://doi.org/10.1073/pnas.1709190114). Proceedings of the National Academy of Sciences of the United States of America. doi:10.1073/pnas.1709190114  
Crema, E.R., Bevan, A. 2021 [Inference from Large Sets of Radiocarbon Dates: Software and Methods](https://doi.org/10.1017/RDC.2020.95) Radiocarbon 63(1),23-39. doi:10.1017/RDC.2020.95

Crema, E.R., Bevan, A., Shennan, S., 2017. [Spatio-temporal approaches to archaeological radiocarbon dates](https://doi.org/10.1016/j.jas.2017.09.007). Journal of Archaeological Science 87, 1–9. doi:10.1016/j.jas.2017.09.007

Crema, E.R., Habu, J., Kobayashi, K., Madella, M., 2016. [Summed Probability Distribution of 14 C Dates Suggests Regional Divergences in the Population Dynamics of the Jomon Period in Eastern Japan](https://doi.org/10.1371/journal.pone.0154809). PLOS ONE 11, e0154809. doi:10.1371/journal.pone.0154809

Edinborough, K., Porcic, M., Martindale, A., Brown, T.J., Supernant, K, Ames, K.M. 2017. [Radiocarbon test for demographic events in written and oral history](https://doi.org/10.1073/pnas.1713012114), Proceedings of the National Academy of Sciences of the United States of America, 114 (47), 12436–12441. doi:10.1073/pnas.1713012114

Timpson, A., Colledge, S., Crema, E., Edinborough, K., Kerig, T., Manning, K., Thomas, M.G., Shennan, S., 2014. [Reconstructing regional population fluctuations in the European Neolithic using radiocarbon dates: a new case-study using an improved method](https://doi.org/10.1016/j.jas.2014.08.011). Journal of Archaeological Science 52, 549–557. doi:10.1016/j.jas.2014.08.011

### Case Studies Using _rcarbon_

Antolín, F., Martínez-Grau, H., Steiner, B. L., Follmann, F., Soteras, R., Häberle, S., et al. 2022. [Neolithic occupations (c. 5200-3400 cal BC) at Isolino Virginia (Lake Varese, Italy) and the onset of the pile-dwelling phenomenon around the Alps](https://doi.org/10.1016/j.jasrep.2022.103375). Journal of Archaeological Science: Reports, 42, 103375. 

Arroyo-Kalin, M., & Riris, P. (2021). [Did pre-Columbian populations of the Amazonian biome reach carrying capacity during the Late Holocene?](https://doi.org/10.1098/rstb.2019.0715) Philosophical Transactions of the Royal Society B: Biological Sciences, 376(1816), 20190715. 

Aubán, J.B., Jiménez-Puerto, J., Ruiz, P.E., Gordo, S.P., 2018. [C14 y poblamiento en las comarcas centro-meridionales del País Valenciano (c. 7000-1500 cal BC)](https://www.raco.cat/index.php/RecerquesMuseuAlcoi/article/view/348376). Recerques del Museu d’Alcoi 0, 35-48–48.

Barnett, R. L., Charman, D. J., Johns, C., Ward, S. L., Bevan, A., Bradley, S. L., et al. 2020. [Nonlinear landscape and cultural response to sea-level rise.](https://doi.org/10.1126/sciadv.abb637) Science Advances, 6(45), eabb6376. 6

Becker, F., Knitter, D., Nykamp, M., Schütt, B., 2020. [Meta-Analysis of Geomorphodynamics in the Western Lower Bakırçay Plain (Aegean Region, Turkey)](https://doi.org/10.3390/land9090338). Land 9, 338. 

Benito, X., Luethje, M., Schneider, T., Fritz, S.C., Baker, P.A., Pedersen, E.J., Gaüzère, P., de Novaes Nascimento, M., Bush, M., Ruhi, A., 2021. [Ecological resilience in tropical Andean lakes: A paleolimnological perspective](https://doi.org/10.1002/lno.11747). Limnology and Oceanography n/a. 

Bergsvik, K. A., Darmark, K., Hjelle, K. L., Aksdal, J., & Åstveit, L. I. (2021). [Demographic developments in Stone Age coastal western Norway by proxy of radiocarbon dates, stray finds and palynological data](https://doi.org/10.1016/j.quascirev.2021.106898). Quaternary Science Reviews, 259, 106898. 

Bevan, A. Colledge,S., Fuller, D., Fyfe, R., Shennan, S., Stevens, C. 2017.[Holocene fluctuations in human population demonstrate repeated links to food production and climate](https://doi.org/10.1073/pnas.1709190114). Proceedings of the National Academy of Sciences of the United States of America, 114 (49) E10524-E10531. doi:10.1073/pnas.1709190114  

Bevan, A., & Crema, E. R. 2021. [Modifiable reporting unit problems and time series of long-term human activity](https://doi.org/10.1098/rstb.2019.0726). Philosophical Transactions of the Royal Society B: Biological Sciences, 376(1816), 20190726. 

Bird, D., Freeman, J., Robinson, E., Maughan, G., Finley, J.B., Lambert, P.M., Kelly, R.L., 2020. [A first empirical analysis of population stability in North America using radiocarbon records](https://doi.org/10.1177/0959683620919975). The Holocene, 30(9), 1345–1359.

Broodbank, C., Lucarini, G. 2019. [The Dynamics of Mediterranean Africa, ca. 9600–1000 bc: An Interpretative Synthesis of Knowns and Unknowns](https://doi.org/10.1558/jma.40581), Journal of Mediterranean Archaeology, 32(2), 195-267.

Broughton,J.M., Weitzel,E.M. 2018. [Population reconstructions for humans and megafauna suggest mixed causes for North American Pleistocene extinctions](https://doi.org/10.1038/s41467-018-07897-1).Nature Communications, 9, 5441. doi:10.1038/s41467-018-07897-1

Brown, A., A., Crema, E. R. 2019. [Māori Population Growth in Pre-contact New Zealand: Regional Population Dynamics Inferred From Summed Probability Distributions of Radiocarbon Dates](https://doi.org/10.1080/15564894.2019.1605429). The Journal of Island and Coastal Archaeology, 1–19. 

Cahiza, P., Garate, E., Sabatini, G., Gheggi, S., Iniesta, M.L., Carosio, S., García, G., 2021. [Temporal dynamics of La Rioja village landscapes, Argentina](https://doi.org/10.1016/j.jasrep.2021.103123). Journal of Archaeological Science: Reports 39, 103123. 

Carter, V. A., Brunelle, A., Power, M. J., DeRose, R. J., Bekker, M. F., Hart, I., et al. (2021). [Legacies of Indigenous land use shaped past wildfire regimes in the Basin-Plateau Region, USA](https://doi.org/10.1038/s43247-021-00137-3). Communications Earth & Environment, 2(1), 1–9. 

Cascalheira, J., Alcaraz-Castaño, M., Alcolea-González, J., de Andrés-Herrero, M., Arrizabalaga, A., Aura Tortosa, J.E., Garcia-Ibaibarriaga, N., Iriarte-Chiapusso, M.-J., 2020. [Paleoenvironments and human adaptations during the Last Glacial Maximum in the Iberian Peninsula: A review.](https://doi.org/10.1016/j.quaint.2020.08.005) Quaternary International 581–582, 28-51.

Cheddadi, R., Palmisano, A., López-Sáez, J. A., Nourelbait, M., Zielhofer, C., Tabel, J., et al. 2019. [Human demography changes in Morocco and environmental imprint during the Holocene](https://doi.org/10.1177/0959683619826657) The Holocene, 0959683619826657. doi:10.1177/0959683619826657

Crema, E.R., Bevan, A. 2020 [Inference from Large Sets of Radiocarbon Dates: Software and Methods](https://doi.org/10.1017/RDC.2020.95) Radiocarbon, doi:10.1017/RDC.2020.95

Crema, E.R., Bevan, A., Shennan, S., 2017. [Spatio-temporal approaches to archaeological radiocarbon dates](https://doi.org/10.1016/j.jas.2017.09.007). Journal of Archaeological Science 87, 1–9. doi:10.1016/j.jas.2017.09.007

Crema, E. R., & Shoda, S. 2021. [A Bayesian approach for fitting and comparing demographic growth models of radiocarbon dates: A case study on the Jomon-Yayoi transition in Kyushu (Japan)](https://doi.org/10.1371/journal.pone.0251695). PLOS ONE, 16(5), e0251695. 

Crema, E.R. 2020 [Non-Stationarity and Local Spatial Analysis](https://doi.org/10.4324/9781351243858), In Gillings, M., Hacıgüzeller, P., Lock,G. (Eds.) Archaeological Spatial Analysis: A Methodological Guide, Taylor & Francis, London.

Crema, E.R., Kobayashi, K., 2020. [A multi-proxy inference of Jōmon population dynamics using bayesian phase models, residential data, and summed probability distribution of 14C dates](https://doi.org/10.1016/j.jas.2020.105136). Journal of Archaeological Science 117, 105136. 

Crema, E. R. 2022. [Statistical Inference of Prehistoric Demography from Frequency Distributions of Radiocarbon Dates: A Review and a Guide for the Perplexed](https://doi.org/10.1007/s10816-022-09559-5). Journal of Archaeological Method and Theory. 

Deforce K., Groenewoudt B., Haneca K. 2020. [2500 years of charcoal production in the Low Countries: The chronology and typology of charcoal kilns and their relation with early iron production.](https://doi.org/10.1016/j.quaint.2020.10.020). Quaternary International.

Deza-Araujo, M., Morales-Molino, C., Conedera, M., Henne, P. D., Krebs, P., Hinz, M., et al. (2022). [A new indicator approach to reconstruct agricultural land use in Europe from sedimentary pollen assemblages](https://doi.org/10.1016/j.palaeo.2022.111051). Palaeogeography, Palaeoclimatology, Palaeoecology, 599, 111051. 

de Groot, B., 2020. [The impact of population fluctuations on the spatial spread of Neolithic ceramic traditions in West Anatolia and South-East Europe.](https://doi.org/10.1016/j.jaa.2019.101121) Journal of Anthropological Archaeology 57, 101121. doi:10.1016/j.jaa.2019.101121

de Groot, B., Bloxam, A., 2022. [Radiocarbon approaches for mapping technological change: The spread of the potter’s wheel in the Iberian Peninsula, 1000–0 BCE](https://doi.org/10.1016/j.jasrep.2021.103288). Journal of Archaeological Science: Reports 41, 103288. 

de Souza, J. G., Robinson, M., Maezumi, S. Y., Capriles, J., Hoggarth, J. A., Lombardo, U., et al. 2019. [Climate change and cultural resilience in late pre-Columbian Amazonia](https://doi.org/10.1038/s41559-019-0924-0). Nature Ecology & Evolution, 1. doi:10.1038/s41559-019-0924-0

de Souza, P., Cartajena, I., Riquelme, R., Maldonado, A., Porras, M. E. de, Santander, B., et al. (2021). [Late Pleistocene–Early Holocene human settlement and environmental dynamics in the southern Atacama Desert highlands (24.0°S–24.5°S, Northern Chile)](https://doi.org/10.1002/gea.21849). Geoarchaeology.

DiNapoli, R. J., Crema, E. R., Lipo, C. P., Rieth, T. M., & Hunt, T. L. (2021). [Approximate Bayesian Computation of radiocarbon and paleoenvironmental record shows population resilience on Rapa Nui (Easter Island)](https://doi.org/10.1038/s41467-021-24252-z). Nature Communications, 12(1), 3939. 

Doering, B. N. 2021. [Subarctic landscape adaptations and paleodemography: A 14,000-year history of climate change and human settlement in central Alaska and Yukon](https://doi.org/10.1016/j.quascirev.2021.107139). Quaternary Science Reviews, 268, 107139. 

Edinborough, K., Martineau, R., Dufraisse, A., Shennan, S., Imbeaux, M., Dumontet, A., et al. (2021). [A Neolithic population model based on new radiocarbon dates from mining, funerary and population scaled activity in the Saint-Gond Marshes region of North East France](https://doi.org/10.1016/j.quaint.2021.03.001). Quaternary International. 

Ekholm, T. 2021. [Hunter-gatherer adaptions during the Early Holocene in Northern Sweden](https://doi.org/10.1177/0959683620961482). The Holocene, 31(1), 83–94. 

Feeser, I., Dörfler, W., Kneisel, J., Hinz, M., & Dreibrodt, S. 2019. [Human impact and population dynamics in the Neolithic and Bronze Age: Multi-proxy evidence from north-western Central Europe](https://doi.org/10.1177/0959683619857223). The Holocene, 0959683619857223. doi:10.1177/0959683619857223

Freeman, J., Anderies, J. M., Beckman, N. G., Robinson, E., Baggio, J. A., Bird, D., et al. (2021). [Landscape Engineering Impacts the Long-Term Stability of Agricultural Populations](https://doi.org/10.1007/s10745-021-00242-z). Human Ecology. 

Freeman, J., Baggio, J.A., Robinson, E., Byers, D.A., Gayo, E., Finley, J.B., Meyer, J.A., Kelly, R.L., Anderies, J.M., 2018. [Synchronization of energy consumption by human societies throughout the Holocene](https://doi.org/10.1073/pnas.1802859115). Proceedings of the National Academy of Sciences of the United States of America, 115 (40) 9962-9967. doi:10.1073/pnas.1802859115

Freeman, J., Hard, R. J., Mauldin, R. P., & Anderies, J. M. (2021). [Radiocarbon data may support a Malthus-Boserup model of hunter-gatherer population expansion](https://doi.org/10.1016/j.jaa.2021.101321). Journal of Anthropological Archaeology, 63, 101321. 

Fritz, C., Tosello, G., Fleury, G., Kasarhérou, E., Walter, P., Duranthon, F., et al. 2021. [First record of the sound produced by the oldest Upper Paleolithic seashell horn](https://doi.org/10.1126/sciadv.abe9510). Science Advances, 7(7), eabe9510. 

Fyfe, R.M., Woodbridge, J., Palmisano, A., Bevan, A., Shennan, S., Burjachs, F., Legarra Herrero, B., García Puchol, O., Carrión, J.-S., Revelles, J., Roberts, C.N., 2019. [Prehistoric palaeodemographics and regional land cover change in eastern Iberia](https://doi.org/10.1177/0959683619826643). The Holocene 0959683619826643. doi:10.1177/0959683619826643

Gil, A.F., Villalba, R., Franchetti, F.R., Otaola, C., Abbona, C.C., Peralta, E.A., Neme, G., 2020. [Between Foragers and Farmers: Climate Change and Human Strategies in Northwestern Patagonia](https://doi.org/10.3390/quat3020017). Quaternary 3, 17. 

Gjesfjeld, E., Etnier, M.A., Takase, K., Brown, W.A., Fitzhugh, B., 2020. [Biogeography and adaptation in the Kuril Islands, Northeast Asia.](https://doi.org/10.1080/00438243.2019.1715248) World Archaeology, 1–25. 

Gjesfjeld, E., Silvestro, D., Chang, J., Koch, B., Foster, J.G., Alfaro, M.E., 2020. [A quantitative workflow for modeling diversification in material culture.](https://doi.org/10.1371/journal.pone.0227579) PLOS ONE 15, e0227579. 

Gretzinger, J., Molak, M., Reiter, E., Pfrengle, S., Urban, C., Neukamm, J., et al. 2019. [Large-scale mitogenomic analysis of the phylogeography of the Late Pleistocene cave bear]( https://doi.org/10.1038/s41598-019-47073-z). Scientific Reports, 9. DOI: 10.1038/s41598-019-47073-z.

Harrison, S.P., Gaillard, M.-J., Stocker, B.D., Vander Linden, M., Klein Goldewijk, K., Boles, O., Braconnot, P., Dawson, A., Fluet-Chouinard, E., Kaplan, J.O., Kastner, T., Pausata, F.S.R., Robinson, E., Whitehouse, N.J., Madella, M., Morrison, K.D., 2020. [Development and testing scenarios for implementing land use and land cover changes during the Holocene in Earth system model experiments](https://doi.org/10.5194/gmd-13-805-2020). Geoscientific Model Development 13, 805–824. 

Hixon, S.W., Douglass, K.G., Godfrey, L.R., Eccles, L., Crowley, B.E., Rakotozafy, L.M.A., Clark, G., Haberle, S., Anderson, A., Wright, H.T., Kennett, D.J., 2021. [Ecological Consequences of a Millennium of Introduced Dogs on Madagascar](https://doi.org/10.3389/fevo.2021.689559). Frontiers in Ecology and Evolution 9, 428. 

Jones, T. L., Coltrain, J. B., Jacobs, D. K., Porcasi, J., Brewer, S. C., Buckner, J. C., et al. (2021). [Causes and consequences of the late Holocene extinction of the marine flightless duck (Chendytes lawi) in the northeastern Pacific](https://doi.org/10.1016/j.quascirev.2021.106914). Quaternary Science Reviews, 260, 106914. 

Jørgensen, E. K., & Riede, F. 2019. [Convergent catastrophes and the termination of the Arctic Norwegian Stone Age: A multi-proxy assessment of the demographic and adaptive responses of mid-Holocene collectors to biophysical forcing.](https://doi.org/10.1177/0959683619862036) The Holocene, 0959683619862036. 

Jorgeson, I.A., Breslawski, R.P., Fisher, A.E., 2020. [Radiocarbon simulation fails to support the temporal synchroneity requirement of the Younger Dryas impact hypothesis](https://doi.org/10.1017/qua.2019.83). Quaternary Research 1–17. 

Kitchel, N.R., Mackie, M.E., 2021. [Plants and Subsistence during the Fluted-Point Period of the Northeast](https://doi.org/10.1017/aaq.2021.125). American Antiquity 1–9. 

Kim, H., Lee, G.-A., Crema, E.R., 2021. [Bayesian analyses question the role of climate in Chulmun demography](https://doi.org/10.1038/s41598-021-03180-4). Scientific Reports 11, 23797. 

Kwak, S., Obata, H., Lee, G.-A., 2020. [Broad-spectrum foodways in southern coastal Korea in the Holocene: Isotopic and archaeobotanical signatures in Neolithic shell middens.](https://doi.org/10.1080/15564894.2020.1776427) The Journal of Island and Coastal Archaeology 0, 1–29. 

Lawrence, D., Palmisano, A., & Gruchy, M. W. de. 2021. [Collapse and continuity: A multi-proxy reconstruction of settlement organization and population trajectories in the Northern Fertile Crescent during the 4.2kya Rapid Climate Change event](https://doi.org/10.1371/journal.pone.0244871). PLOS ONE, 16(1), e0244871. 

Lima, M., Gayo, E.M., Latorre, C., Santoro, C.M., Estay, S.A., Cañellas-Boltà, N., Margalef, O., Giralt, S., Sáez, A., Pla-Rabes, S., Chr. Stenseth, N., 2020. [Ecology of the collapse of Rapa Nui society](https://doi.org/10.1098/rspb.2020.0662). Proceedings of the Royal Society B: Biological Sciences 287, 20200662. 

Liu, L., Chen, X., Wright, H., Xu, H., Li, Y., Chen, G., et al. 2019. [Rise and fall of complex societies in the Yiluo region, North China: The spatial and temporal changes.](https://doi.org/10.1016/j.quaint.2019.05.025) Quaternary International. 

López, J. M., Neme, G., & Gil, A. F. 2019. [Resource intensification and zooarchaeological record in the southern margins of pre-Hispanic Andean agriculture.](https://doi.org/10.1007/s12520-019-00857-w) Archaeological and Anthropological Sciences, 11(10), 5287–5300. 

Lucarini, G., Wilkinson, T., Crema, E.R., Palombini, A., Bevan, A., Broodbank, C., 2020. [The MedAfriCarbon Radiocarbon Database and Web Application. Archaeological Dynamics in Mediterranean Africa, ca. 9600–700 BC.](https://doi.org/10.5334/joad.60) Journal of Open Archaeology Data 8, 1. 

MacInnes, D. 2019. [The impact of population dynamics on social complexity in Neolithic Orkney.](https://doi.org/10.1016/j.jasrep.2019.02.036) Journal of Archaeological Science: Reports, 24, 721–728. doi:10.1016/j.jasrep.2019.02.036

McCool, W.C., Tung, T.A., Coltrain, J.B., Accinelli Obando, A.J., Kennett, D.J., 2021. [The character of conflict: A bioarchaeological study of violence in the Nasca highlands of Peru during the Late Intermediate Period (950–1450 C.E.)](https://doi.org/10.1002/ajpa.24205). American Journal of Physical Anthropology 174, 614–630. 


NEWS.md
# Version 1.5.1 (in development)
* Bug Fix: Removed non-relevant warning messages from `calibrate()` within `modelTest()`.
* NEW: `qCal()` function computes quantiles of calibrated dates.
* UPDATE: `multiplot()` now enables relative positioning of labels based on quantiles of calibrated dates and/or a user-defined offset. 

# Version 1.5.0 (21 November 2022)
* UPDATE: Functions related to spatial permutation test (`spweights()`,`sptest()`, and `plot.spatialTest()`) now depend on `sf` instead of `sp` package.
* UPDATE: `modelTest()` now issues a warning message when `model` is set to `"exponential"` or `"uniform"` (i.e. when the SPD is tested against exponential and uniform curves fitted on observed SPD values.) 
* UPDATE: Updated vignette with minor changes reflecting the workflow for `sptest()` and the warning in `modelTest()`.
* Bug Fix: Fixed a problem with `plot.stackCalSPD()` function when attempting to display proportion SPD plots including intervals with no dates.
* Updated spatstat dependencies.

# Version 1.4.4 (21 July 2022)
* Bug Fix: Fixed a bug in `spd()` reducing the `timeRange` when `datenormalised` was set to TRUE.
* UPDATE: `calibrate()` now issues a warning in case dates have calibrated probabilities outside the user defined `timeRange` value.
* UPDATE: `stackspd()` now requires the `group` argument to be factors. Factor levels will be condition the plot order in `plot.stackCalSPD()`.
* UPDATE: when setting the `rescale=TRUE` and `type='multipanel'`, `plot.stackCalSPD()` will now display the true y-axis range of individual SPDs (instead of a rescaled 0-1 range).   

# Version 1.4.3 (17 February 2022)
* UPDATE: Reversed legend item order in `plot.stackCalSPD()` to match display
* UPDATE: `hpdi()` now has the option to return a character vector with HPDI ranges in either BP or BCAD.

# Version 1.4.2 (15 March 2021)
* Bug fix: Fixed a number of minor bugs and errors caused when the summed probability vector contains zero. Functions affected by this bugs were:
  * proportion plots in `plot.stackCalSPD()`
  * `summary()` and `plot()` functions for `SpdModelTest` class objects
* Bug Fix: Fixed issues with `summary()` function on `CalDates` class object with multiple dates (thanks to [nferebau](https://github.com/nfrerebeau) for the bug report and fix)
`plot.stackCalSPD()` proportion plot to handle instances with 0 summed probability in the time Range of analysis.
* UPDATE: `binMed()` function can now handle larger `caldates` objects.
* NEW: `poolDates()` function allows combination of 14C ages associated with the same event using Ward and Wilson 1978 method.
* Minor bug fixes.
* Minor typos and errors in the help documentation.
* Updated dependencies for `spatstat(>= 2.0-0)`. 


# Version 1.4.1 (6 October 2020)
* UPDATE: Improved performance of the `calibrate()` function (c.a 400% faster)
* UPDATE: `multiplot` allows for rescaled calibrated probability distribution (new argument `rescale`) for improved readability.
* UPDATE: The package vignette now includes examples from new functions such as `multiplot()` and `stackspd()`.
* UPDATE: `hdmi()` now computes the probability mass of each age bracket (thanks to [hanecakr](https://github.com/hanecakr) for the suggestion and sample code)
* NEW: `transformSPD()` allows for taphonomic (and other user-defined) corrections to SPDs.
* Bug fix: Fixed color matching and improved label placing  in the `multiplot` function.
* Bug fix: calCurve='normal' no longer causing error in `calibrate()`

# Version 1.4.0 (15 August 2020)
* IntCal20, ShCal20, and Marine20 curves added, with IntCal20 as default calibration curve for `calibrate()`.

# Version 1.3.3 (4 August 2020)
* Minor changes on NAMESPACE and test environment to ensure workable CRAN checks for all operating systems.

#  Verson 1.3.2 (25 July 2020)
* Bug fix : `F14C=TRUE` was producting an error message in `calibrate()` after refactoring in version 1.3.1. 
* Bug fix : fixed error in the handling of `resErrors` (Delta R error) for  marine reservoir effect.
* NEW: `stackspd()` creates a set of multiple SPDs as an object of class `stackCalSPD` with a dedicated plot function.
* NEW: `multiplot()` function for displaying multiple calibrated dates.
* NEW: `hpdi()` function for computing highest probability density intervals of calibrated dates.
* NEW: `combine()` function for concatenating multiple `calDates` class objects.
* UPDATE: `calibrate()`, `modelTest()`, and `sptest()` are now parallelised using the *doSNOW* package, enabling progress bar when running over multiple cores.

#  Version 1.3.1 (18 March 2020)
* `binPrep()` now accepts alternative clustering algorithms
* Refactoring of calibration related functions to increase readability
* Minor bug fixes

#  Version 1.3.0 (12 December 2019)
* UPDATEs in `modelTest()`:
  * Added an error message and a note in the help documentation warning users to not use calibration curves not supported by `uncalibrate()`.
  * NEW argument `gridclip` to add range constrain in the simulation envelope. When set to TRUE the sampling is executed within the observed range of dates.
  * UPDATE `timeRange` can now be left undefined, in which case the range of median calibrated observed dates is used. 
  * NEW argument `normalised` handles whether simulated dates should be normalised or not.
  * NEW arguments `backsight` and `changexpr` enabling the comparison of expected and observed rates of changes. 
  * UPDATE: improved performance of `calsample` and `uncalsample` methods (update in `uncalibrate()`) `
  * Bug fix: users no longer need to specify `nsim` when `fitonly` is set to TRUE.
* UPDATE in `permTest()`
  * NEW arguments `backsight` and `changexpr` enabling the comparison of expected and observed rates of changes. 
* UPDATE in `spd()`: aggregation matrix now spans beyond `timeRange` to avoid edge effects.
* UPDATE in `calibrate()`: the new argument `F14C` enables calibration in F14C space.
* UPDATE in `SPpermTest()`: 

R/data.R
#' @title Radiocarbon dates from the EUROEVOL database
#' @description Radiocarbon dates (n=14,053) and site coordinates (n=4,213) from the EUROEVOL project database. Sites without radiocarbon dates (n=544), phase-codes, and other data have been omitted.

#' @format A data.frame with the following variables:
#' \describe{
#' \item{\code{C14ID}}{ID of each radiocarbon date}
#' \item{\code{C14Age}}{Radiocarbon age in 14C years BP}
#' \item{\code{C14SD}}{Radiocarbon age error}
#' \item{\code{LabCode}}{Labcode of the radiocarbon date}
#' \item{\code{Material}}{Material of the dated sample}
#' \item{\code{SiteID}}{ID of the site from which the sample has been recovered}
#' \item{\code{Latitude}}{Latitude of the sampling site in decimal degrees}
#' \item{\code{Longitude}}{Longitude of the sampling site in decimal degrees}
#' \item{\code{Country}}{Country where the sampling site is located}
#'}
#' @source Manning, K., Colledge, S., Crema, E., Shennan, S., Timpson, A., 2016. The Cultural Evolution of Neolithic Europe. EUROEVOL Dataset 1: Sites, Phases and Radiocarbon Data. Journal of Open Archaeology Data 5. doi:10.5334/joad.40
#' @references
#' Shennan, S., Downey, S.S., Timpson, A., Edinborough, K., Colledge, S., Kerig, T., Manning, K., Thomas, M.G., 2013. Regional population collapse followed initial agriculture booms in mid-Holocene Europe. Nature Communications 4, ncomms3486. doi:10.1038/ncomms3486
#'
#' Timpson, A., Colledge, S., Crema, E., Edinborough, K., Kerig, T., Manning, K., Thomas, M.G., Shennan, S., 2014. Reconstructing regional population fluctuations in the European Neolithic using radiocarbon dates: a new case-study using an improved method. Journal of Archaeological Science 52, 549-557. doi:10.1016/j.jas.2014.08.011
#'
#' @examples
#' \dontrun{
#' data(euroevol)
#' Ireland <- subset(euroevol,Country=="Ireland")
#' bins <- binPrep(Ireland$SiteID,Ireland$C14Age,h=200)
#' x <- calibrate(Ireland$C14Age,Ireland$C14SD)
#' spd.ireland <- spd(x,bins=bins,runm=200,timeRange=c(8000,4000))
#' plot(spd.ireland)
#'}
"euroevol"

#' @title Radiocarbon dates for the Eastern Mediterranean around the Younger Dryas
#' @description Radiocarbon dates (n=1915) and site coordinates (n=201) from a paper considering the relationship between human activity in the eastern Mediterranean/Middle East and early Holocene climate change, including the Younger Dryas.   

#' @format A data.frame with the following variables:
#' \describe{
#' \item{\code{LabID}}{Laboratory ID assigned to each radiocarbon date (where known)}
#' \item{\code{CRA}}{Radiocarbon age in 14C years BP}
#' \item{\code{Error}}{Radiocarbon age error}
#' \item{\code{Material}}{Material of the dated sample}
#' \item{\code{Species}}{Species of the dated sample (where identified)}
#' \item{\code{SiteName}}{Name of the site from which the sample has been recovered}
#' \item{\code{Country}}{Country where the sampling site is located}
#' \item{\code{Longitude}}{Longitude of the sampling site in decimal degrees}
#' \item{\code{Latitude}}{Latitude of the sampling site in decimal degrees}
#' \item{\code{Region}}{One of three analytical regions (1=southern Levant, 2=Northern Levant, 3= South-central Anatolia}
#'}
#' @source Palmisano, A., Bevan, A. and S. Shennan 2017. Data and code for demographic trends in the paper "Human responses and non-responses to climatic variations during the Last Glacial-Interglacial transition in the eastern Mediterranean", UCL Discovery Archive 1570274. doi:10.14324/000.ds.1570274.
#' @references
#' Roberts, N., Woodbridge, J., Bevan, A., Palmisano, A., Shennan, S. and E. Asouti 2017. Human responses and non-responses to climatic variations during the Last Glacial-Interglacial transition in the eastern Mediterranean. Quaternary Science Reviews, 184, 47-67. doi:10.1016/j.quascirev.2017.09.011.
#' 
#' @examples
#' \dontrun{
#' data(emedyd)
#' northernlevant <- emedyd[emedyd$Region=="2",]
#' bins <- binPrep(northernlevant$SiteName, northernlevant$CRA, h=50)
#' x <- calibrate(northernlevant$CRA, northernlevant$Error, normalised=FALSE)
#' spd.northernlevant <- spd(x, bins=bins, runm=50, timeRange=c(17000,8000))
#' plot(spd.northernlevant)
#'}
"emedyd"


#' @title Subset of EUROEVOL radiocarbon dates from Great Britain
#' @description Radiocarbon dates (n=2,324) and site coordinates (n=652) from England and Wales collected from the EUROEVOL project database. See \cite{\link{euroevol}} for more details regarding the source data.
#'
#' @format A data.frame with the following variables:
#' \describe{
#' \item{\code{C14ID}}{ID of each radiocarbon date}
#' \item{\code{C14Age}}{Radiocarbon age in 14C years BP}
#' \item{\code{C14SD}}{Radiocarbon age error}
#' \item{\code{LabCode}}{Labcode of the radiocarbon date}
#' \item{\code{Material}}{Material of the dated sample}
#' \item{\code{SiteID}}{ID of the site from which the sample has been recovered}
#' \item{\code{Eastings}}{Easting coordinates of the sampling site in meters (OSGB 1936 epsg:27700) }
#' \item{\code{Northings}}{Northing coordinates of the sampling site in meters (OSGB 1936 epsg:27700)}
#'}
#' @source Manning, K., Colledge, S., Crema, E., Shennan, S., Timpson, A., 2016. The Cultural Evolution of Neolithic Europe. EUROEVOL Dataset 1: Sites, Phases and Radiocarbon Data. Journal of Open Archaeology Data 5. doi:10.5334/joad.40
#'
"ewdates"


#' @title Polygonal window of England and Wales 
#' @description An \code{\link{owin}} class polygonal window of England and Wales.
#' @format An \code{owin} class object. 
#' @examples
#' \dontrun{
#' data(ewowin)
#' # Obtained from rworldmap:
#' library(maptools)
#' library(rgeos)
#' library(rworldmap)
#' bng <- CRS("+init=epsg:27700")
#' sbrit <- getMap(resolution="high")
#' sbrit <- spTransform(sbrit[sbrit$GEOUNIT =="United Kingdom",], bng)
#' tmp <- SpatialPolygons(list(Polygons(list(Polygon(cbind(c(130000,130000,310000,
#' 425000,700000,700000,130000), c(0,300000,560000,598000,300000,0,0)))), 
#' "1")), 1:1, proj4string=bng)
#' sbrit <- gIntersection(sbrit,tmp) 
#' ewowin <- as.owin(sbrit)
#' }
#' @source South,A.2011.rworldmap: A New R package for Mapping Global Data. The R Journal Vol. 3/1 : 35-43.
"ewowin"

